src/Pulp/Run.purs

```diff
@@ -60,6 +61,33 @@ jsEscape =
   replace (Pattern "'") (Replacement "\\'") <<<
   replace (Pattern "\\") (Replacement "\\\\")
 
+makeRunnableScript :: { out :: Outputter, buildPath :: String, prefix :: String, moduleName :: String } -> Aff String
+makeRunnableScript { out, buildPath, prefix, moduleName } = do
+  psVer <- getPursVersion out
+  fullPath' <- liftEffect $ Path.resolve [] buildPath
+  let
+    fullPath = replaceAll (Pattern "\\") (Replacement "/") fullPath'
+    { makeEntry, suffix } =
+      if (dropPreRelBuildMeta psVer) < psVersions.v0_15_0 then
+        { makeEntry: makeCjsEntry, suffix: ".js" }
+      else
+        { makeEntry: makeEsEntry fullPath, suffix: ".mjs" }
```

Comment:
Is the `.mjs` suffix required? What is Spago doing for its script?

Reply:
Well, here's our options for this.

1. We call `node` using the `--eval` arg and it evaluates the string you give as a script. To make that work on ES modules, we include the `--input-type module` flag, so that the whole code looks like this: `node --input-type module --eval "import { main } from 'file:///path/to/rootDir/output/Module.Name/index.js'; main();"`. Unfortunately, this is problematic as described in [an earlier comment](https://github.com/purescript-contrib/pulp/pull/401#issuecomment-1059610898)
1. We use `.js` and have to also add a `package.json` file that has `{"type": "module"}` as its contents in the directory containing it, so that Node interprets that JS file as ES modules rather than CommonJS files. AFAIK, this is likely fine since `purs` creates that 'package.json` file
1. We use `.mjs` because Node will immediately understand that it's ES modules, regardless of whether the `package.json` file is there.

I went with 3 mainly because it removed an additional problem that could arise. Now that CI builds, I think it's worth it to try the `.js` version.

Reply:
If we can do just the `.js` version I'd prefer that.

Reply:
Welp. Looks like I might be wrong about 3. https://github.com/purescript-contrib/pulp/runs/5488535135?check_suite_focus=true#step:8:583

Reply:
Yup, the `package.json` file is there, but perhaps `node` isn't treating it as ES modules because the root dir's `package.json` file doesn't have `type: module`? I'm going to revert this back to `.mjs` because that at least builds.

Reply:
I took a look at this again. I think the issue is that we're not passing in the `--experimental-modules` flag to Node when Node is < `13.0.0`. See https://github.com/working-group-purescript-es/purescript/pull/11 where this same issue arose. 

So, I installed Node 12.6.0 and ran a `pulp run` test locally. However, I still got an error. So, I tried out a simple example to see if there was something else going on:
```
$ cat psOutput.js 
export const main = function () {
  console.log("this works");
}
$ cat entryPoint.js 
import { main } from '/home/jordan/Programming/Projects/pulp/psOutput.js';

main();
$ node --version
v12.6.0
$ node --experimental-modules entryPoint.js
(node:28551) ExperimentalWarning: The ESM module loader is experimental.
/home/jordan/Programming/Projects/pulp/entryPoint.js:1
import { main } from 'file:///home/jordan/Programming/Projects/pulp/entryPoint.js';
       ^

SyntaxError: Unexpected token {
    at Module._compile (internal/modules/cjs/loader.js:720:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at internal/modules/esm/translators.js:87:15
    at Object.meta.done (internal/modules/esm/create_dynamic_module.js:48:9)
    at file:///home/jordan/Programming/Projects/pulp/entryPoint.js:9:13
    at ModuleJob.run (internal/modules/esm/module_job.js:111:37)
    at processTicksAndRejections (internal/process/task_queues.js:85:5)
    at async Loader.import (internal/modules/esm/loader.js:134:24)
```
I tried `import Main ...` and `import * as Main ...`, and still got the same error.

Reply:
I shifted the files around a bit
```sh
$ ls z/
entryPoint.js  package.json  psOut.js
$ cat z/package.json 
{"type": "module"}
$ node --experimental-modules z/entryPoint.js 
(node:29562) ExperimentalWarning: The ESM module loader is experimental.
this works
```
So, using `.js` can work. The issue is the `package.json` file's location.

Reply:
OK! Figured it out. The issue is that the script generated by `pulp` is stored in the `/tmp/` directory. Since that directory doesn't contain a `package.json` file with `{"type": "module"}` as its content, Node throws the above error. Once I create a `package.json` file there, Node runs fine.